cosmos/dbt/selector.py

```diff
@@ -85,26 +89,27 @@ def select_nodes_ids_by_intersection(nodes: dict[str, DbtNode], config: Selector
     """
     selected_nodes = set()
     for node_id, node in nodes.items():
-        if config.tags and not (sorted(node.tags) == sorted(config.tags)):
-            continue
-
-        supported_node_config = {key: value for key, value in node.config.items() if key in SUPPORTED_CONFIG}
-        if config.config:
-            config_tag = config.config.get("tags")
-            if config_tag and config_tag not in supported_node_config.get("tags", []):
+        if not config.is_empty:
+            if config.tags and not (sorted(node.tags) == sorted(config.tags)):
                 continue
 
-            # Remove 'tags' as they've already been filtered for
-            config.config.pop("tags", None)
-            supported_node_config.pop("tags", None)
+            supported_node_config = {key: value for key, value in node.config.items() if key in SUPPORTED_CONFIG}
+            if config.config:
+                config_tag = config.config.get("tags")
+                if config_tag and config_tag not in supported_node_config.get("tags", []):
+                    continue
 
-            if not (config.config.items() <= supported_node_config.items()):
-                continue
+                # Remove 'tags' as they've already been filtered for
+                config.config.pop("tags", None)
+                supported_node_config.pop("tags", None)
 
-        if config.paths and not (set(config.paths).issubset(set(node.file_path.parents))):
-            continue
+                if not (config.config.items() <= supported_node_config.items()):
+                    continue
```

Comment:
After the first match, we'd lose this configuration when we iterate over the following nodes, and nodes that should not match, may match because of the missing of config. Suggestion:
```
                config_copy = copy.deepcopy(config.config)
                config_copy.pop("tags", None)
                supported_node_config.pop("tags", None)

                if not (config_copy.items() <= supported_node_config.items()):
                    continue
```
Sample code:
https://github.com/astronomer/astronomer-cosmos/commit/5b961a5590cc7f512ea8e8e3959c732687cc30e4

I ran:
```
hatch run tests.py3.8-2.6:test-cov
```

And things seem to work as expected 🤞

Reply:
[9138cf7](https://github.com/astronomer/astronomer-cosmos/pull/463/commits/9138cf7013df9153ac680b5de3c2f4fba7bdacb3)